### PR TITLE
Rework directory structure

### PR DESCRIPTION
**Pull request by ajhenry**
This does a lot so hold onto your 🎩s 

- Renamed backend and frontend folders to `backend` and `app` respectively
- Hoisted env files to root
- Renamed backend `start` to `dev`
- Deleted old go backend
- Reworked deploy workflow
- Setup npm workspaces

### CODE REVIEW

**Review by Lehcar — APPROVED** [review]

**Review by ipc103 — APPROVED** [review]
Changes are mostly related to:

- renaming files
- removing Tremor
- Running prettier/eslint

Change LGTM! I'll test Codespaces once this merges just to make sure we're still in good shape there 👍

File: .vscode/settings.json
Comment (ajhenry) [reply .vscode/settings.json:3]: Yes :)

File: app/.eslintrc.js
Comment (ajhenry) [reply app/.eslintrc.js:1]: This was here when I joined the project but has lots of github related rules

**Pull request merged**
